src/main/java/com/github/dockerjava/core/command/CreateContainerCmdImpl.java

```diff
@@ -125,11 +139,50 @@ public CreateContainerCmdImpl(CreateContainerCmd.Exec exec, String image) {
      */
     @Override
     public CreateContainerResponse exec() throws NotFoundException, ConflictException {
+        //code flow taken from https://github.com/docker/docker/blob/master/runconfig/opts/parse.go
```

Comment:
So what is your suggestion to handle this properly? The problem is, that links must be set on different objects, whenever the link is made on the old legacy network or on a new user defined network.